### DISCUSSION TITLE
Question regarding extractSignatures command

### DISCUSSION THREAD

**Discussion by ghost** [PoisonAlien/maftools#645] [Q&A] [answered]
Hi @PoisonAlien, 

maftools package is brilliant. Have a small question about `extractSignatures` though. Is there any particular reason for absence of nrun argument with this function? It seems to be already in use with `estimateSignatures`.  

Since you are using `nmf` underneath, only 1 run is performed by default for the chosen rank. Alternative `extract_signatures` function from MutationalPatterns package already uses 200 runs by [default](https://rdrr.io/bioc/MutationalPatterns/man/extract_signatures.html).

**Comment by PoisonAlien** [comment, marked as answer]
Hello,
Thanks for using `maftools` and I am glad that you find it useful. 
In my experience `nrun` has an insignificant influence on the overall results. The critical step is `estimateSignatures` and as long as you have this approximation, it should be fine for the main `extractSignatures`.  `nrun` indeed helps when the data has lots of observation (as in DNAm or gene expression), since here there are only 96 channels, it should not affect the results much. 

Also, you are welcome to test this by adding `nrun = 100` to [this line](https://github.com/PoisonAlien/maftools/blob/master/R/extractSignatures.R#L60) of the `extractSignatures` script :)

Hope this helps.

**Reply by ghost** [reply]
Makes sense. Thank you.

**Comment by ShixiangWang** [comment]
There is a function `sig_extract()` in **sigminer** which is similar to `extractSignatures()` in **maftools**, you can directly use it.

[[doc](https://shixiangwang.github.io/sigminer/reference/sig_extract.html), [source code](https://github.com/ShixiangWang/sigminer/blob/master/R/sig_extract.R)]

**Reply by ghost** [reply]
Thank you. I was actually checking out sigminer last night and might just use this function. For now though, I have added the `nrun` argument to `extractSignatures` with 50 runs by default, and an argument to get the absolute exposures as well. By default, `extractSignatures` only returns the adjusted exposures.
